src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-wizard/rgw-multisite-wizard.component.ts

```diff
@@ -0,0 +1,258 @@
+      .list()
+      .pipe(
+        switchMap((daemons) => {
+          this.daemons = daemons;
+          const daemonStatsObservables = daemons.map((daemon) =>
+            this.rgwDaemonService.get(daemon.id).pipe(
+              map((daemonStats) => ({
+                hostname: daemon.server_hostname,
+                port: daemon.port,
+                frontendConfig: daemonStats['rgw_metadata']['frontend_config#0']
+              }))
+            )
+          );
+          return forkJoin(daemonStatsObservables);
+        })
+      )
+      .subscribe((daemonStatsArray) => {
+        this.rgwEndpoints.value = daemonStatsArray.map((daemonStats) => {
+          const protocol = daemonStats.frontendConfig.includes('ssl_port') ? 'https' : 'http';
+          return `${protocol}://${daemonStats.hostname}:${daemonStats.port}`;
+        });
+        const options: SelectOption[] = this.rgwEndpoints.value.map(
+          (endpoint: string) => new SelectOption(false, endpoint, '')
+        );
+        this.rgwEndpoints.options = [...options];
+      });
+
+    this.multiClusterService.getCluster().subscribe((clusters) => {
+      this.clusterDetailsArray = Object.values(clusters['config'])
+        .flat()
+        .filter((cluster) => cluster['cluster_alias'] !== 'local-cluster');
+      this.isMultiClusterConfigured = this.clusterDetailsArray.length > 0;
+      if (!this.isMultiClusterConfigured) {
+        this.stepTitles = ['Create Realm & Zonegroup', 'Create Zone', 'Export Multi-site token'];
+      } else {
+        this.selectedCluster = this.clusterDetailsArray[0]['name'];
+      }
+    });
+  }
+
+  createForm() {
+    this.multisiteSetupForm = new CdFormGroup({
+      realmName: new UntypedFormControl('default_realm', {
+        validators: [Validators.required]
+      }),
+      zonegroupName: new UntypedFormControl('default_zonegroup', {
+        validators: [Validators.required]
+      }),
+      zonegroup_endpoints: new UntypedFormControl(null, [Validators.required]),
+      zoneName: new UntypedFormControl('default_zone', {
+        validators: [Validators.required]
+      }),
+      zone_endpoints: new UntypedFormControl(null, {
+        validators: [Validators.required]
+      }),
+      username: new UntypedFormControl('default_system_user', {
+        validators: [Validators.required]
+      }),
+      cluster: new UntypedFormControl(null, {
+        validators: [Validators.required]
+      })
+    });
+
+    if (!this.isMultiClusterConfigured) {
+      this.exportTokenForm = new CdFormGroup({});
+    }
+  }
+
+  showSubmitButtonLabel() {
+    if (this.isMultiClusterConfigured) {
+      return !this.wizardStepsService.isLastStep()
+        ? this.actionLabels.NEXT
+        : $localize`Configure Multi-site`;
+    } else {
+      return !this.wizardStepsService.isLastStep() ? this.actionLabels.NEXT : $localize`Close`;
+    }
+  }
+
+  showCancelButtonLabel() {
+    return !this.wizardStepsService.isFirstStep()
```

Comment:
NIT

Not a fan of the added complexity here, why the need to invert every value returned by `this.wizardStepsService`


By adding complexity without necessity we could end up with things like: `!this.wizardStepsService.isNotLastStep()` in the future

Simpler is easier to read

Reply:
Doesn't seem to be very informative `isNotLastStep` because doesn't specify exactly which step you are on.